chainer/links/loss/hierarchical_softmax.py

```diff
@@ -349,6 +394,64 @@ def create_huffman_tree(word_counts):
 
         return q.get()[2]
 
+    def sample(self, x):
+        """Samples an example for a given input from the tree.
```

Comment:
Could you elabolate what this method does? In which distribution the sample is taken? From code example, the result does not seem to be an input data.